common/core/hwclayerbuffermanager.cpp

```diff
@@ -0,0 +1,74 @@
+/*
+// Copyright (c) 2016 Intel Corporation
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+*/
+
+#include "hwclayerbuffermanager.h"
+#include <hwclayer.h>
+
+#include <cmath>
+
+namespace hwcomposer {
+
```

Comment:
Can you add documentation as how this manager would work i.e. how it would release the cached buffers etc

Reply:
OK. I think the whole design is not complicate. So I list a few key points here:
1: the hwclayerbuffermanager is owned per display, as each display has a separate GL context 
2: hwclayerbuffermanager stores refernces of external buffers in a vector, each vector member is hash map. The vector stores history buffer in this way, vector[0] is for the current frame buffers, vector[1] is for last frame (-1) buffers, vector[2] is -2 frames buffers etc. A constant (currently 4) frames of buffers is stored. When a buffer refernce  is stored in vector[3] and it is not used in the current present, it will go out of scope and be released. If a buffer is fetched from map, it will always re-registered in vector[0] map.
3; egl-image and texture can be owned by drmbuffer and reused

Reply:
Adding it as comments in Code would help for future.